docs/ci/exceptions.rst

```diff
@@ -0,0 +1,37 @@
+nbcollection-ci Exceptions
```

Comment:
This is very verbose and cool! But I wonder if we need the examples? One option would be to remove the examples and give clearer explanations about specifically these various errors will arise. But another possibility would be to just remove the examples and keep this page?